PayloadPkg/OsLoader/OsLoader.c

```diff
@@ -705,7 +827,11 @@ PayloadMain (
   //
   mCurrentBoot = GetCurrentBootOption (OsBootOptionList, mCurrentBoot);
   while  (mCurrentBoot < OsBootOptionList->OsBootOptionCount) {
-    BootOsImage (&OsBootOptionList->OsBootOption[mCurrentBoot]);
+    if((OsBootOptionList->OsBootOption[mCurrentBoot].BootFlags & BOOT_FLAGS_PRE_BOOT) != 0) {
+      BootPreBootImage (&OsBootOptionList->OsBootOption[mCurrentBoot], &OsBootOptionList->OsBootOption[mCurrentBoot+1]);
```

Comment:
The pre-OS and OS image might be stored on different media devices so here we pass in the boot options individually if BOOT_FLAGS_PRE_BOOT type is detected. Also, the pre-OS and OS image are expected to be at index=n and index=n+1 respectively.

Reply:
Do we really have requirement to boot pre-OS and OS in different boot media? By our current design, our boot option doesn't support this case. It would be easier to have them both in same boot media.

Reply:
Yes. There is a request for the option to integrate and launch an IFWI/SPI IAS pre-OS image and load an OS image from another storage media device that the pre-OS image will launch after it finishes execution.